### ISSUE TITLE
libfmod.so still creating trouble in QtCreator linux

### ISSUE THREAD

**Issue by artificiel** [openframeworks/openFrameworks#7462]
downloading a fresh linux64 nightly (or git+download_libs) and trying to compile with QtCreator leads to some errors that are fixed by #7461, but then fails with the `libfmod.so.12: cannot open shared object file` error (refs: #6733 and [build-run-from-qtcreator-libfmod-so-12](https://forum.openframeworks.cc/t/build-run-from-qtcreator-libfmod-so-12/37565))

(to be clear the same project/example compiles and run with `make && make run`.)

the suggestion [here](https://forum.openframeworks.cc/t/build-run-from-qtcreator-libfmod-so-12/37565/7?u=burton) (excluding the lib) works, but @ofTheo mentions it not being an "ideal fix" — in which way is it not ideal? 

note that running [scripts/linux/removeFMOD.sh](https://github.com/openframeworks/openFrameworks/blob/master/scripts/linux/removeFMOD.sh) also fixes the problem — is it the plan anyway to drop fmod? qbs is finding the library by itself (it is not explicitly included/linked).

so is it (a) the "not ideal active exclusion", (b) fmod is removed from the libs or (c) something else?

**Comment by ofTheo** [comment]
@artificiel I think adding fmod to the lib excludes could be fine. 
https://github.com/openframeworks/openFrameworks/blob/master/libs/openFrameworksCompiled/project/qtcreator/modules/of/of.qbs#L135-L146

I am guessing the better fix would be to make sure shared libs are copied to the build directory:
https://github.com/openframeworks/openFrameworks/blob/master/libs/openFrameworksCompiled/project/qtcreator/modules/of/helpers.js#L145-L147 

If you copy the contents of fmod/lib/linux64/ to bin/ does it work for you?

**Comment by artificiel** [comment]
it does not find it when in ./bin besides the executable -- it's looking for libfmod.so.12, and a symlink or plain rename does not work.
```
[artificiel@assez-puissant bin]$ ls -l
total 34576
drwxr-xr-x 2 artificiel wheel     4096 Apr  2 00:49 data
-rwxr-xr-x 1 artificiel wheel  4147408 Apr  3 22:20 freshgit
-rwxr-xr-x 1 artificiel wheel 29614552 Apr  2 00:55 freshgit_debug
-rwxr-xr-x 1 artificiel wheel  1627616 Apr  3 22:16 libfmod.so
lrwxrwxrwx 1 artificiel wheel       40 Apr  3 22:19 libfmod.so.12 -> /home/artificiel/freshgit/bin/libfmod.so
```
```
[artificiel@assez-puissant bin]$ ./freshgit
./freshgit: error while loading shared libraries: libfmod.so.12: cannot open shared object file: No such file or directory
```
``` 
[artificiel@assez-puissant bin]$ ldd freshgit
linux-vdso.so.1 (0x00007ffd155fa000)
libfmod.so.12 => not found
libgstapp-1.0.so.0 => /usr/lib/libgstapp-1.0.so.0 (0x00007f7729a67000)
libgstvideo-1.0.so.0 => /usr/lib/libgstvideo-1.0.so.0 (0x00007f77299a3000)
libgstbase-1.0.so.0 => /usr/lib/libgstbase-1.0.so.0 (0x00007f772991e000)
... etc
```
it does however work if libfmod.so.12 is copied into /usr/lib ...
```
[artificiel@assez-puissant bin]$ ldd freshgit
linux-vdso.so.1 (0x00007ffc90cd8000)
libfmod.so.12 => /usr/lib/libfmod.so.12 (0x00007f6b2b200000)
libgstapp-1.0.so.0 => /usr/lib/libgstapp-1.0.so.0 (0x00007f6b2b643000)
libgstvideo-1.0.so.0 => /usr/lib/libgstvideo-1.0.so.0 (0x00007f6b2b13c000)
libgstbase-1.0.so.0 => /usr/lib/libgstbase-1.0.so.0 (0x00007f6b2b0b7000)
... etc
```

**Comment by artificiel** [comment]
also, i am not sure how this happened, another git-based OF tree (on the same linux machine) that been cloned and updated since a few years contains not libfmod but libfmodex, which works correctly. i don't remember doing specific things around that.

**Comment by ofTheo** [comment]
> it does not find it when in ./bin besides the executable -- it's looking for libfmod.so.12, and a symlink or plain rename does not work.

Thanks for looking into this - looks like the LD_LIBRARY_PATH or rpath might need to be set to load the lib relative to the executable. ( hence why it works fine in /usr/lib ) - see: https://serverfault.com/questions/279068/cant-find-so-in-the-same-directory-as-the-executable 

Happy to disable it for now via:
https://github.com/openframeworks/openFrameworks/blob/master/libs/openFrameworksCompiled/project/qtcreator/modules/of/of.qbs#L135-L146 

Or if you want to take a stab at setting an rpath? ( no idea how to do that in QTCreator )

**Comment by artificiel** [comment]
setting the rpath to include the binary's dir is straightforward: `cpp.rpaths: cpp.rpathOrigin`. however copying a lib as a post-compile step within the .qbs (not .pro) seems complicated. considering qbs is deprecated anyway, and that furthermore the audio player is correctly detected as OpenAL, and that it works, i think excluding fmod is a reasonable thing to do.

**Comment by ofTheo** [comment]
@artificiel excluding sounds good to me too 👍

**Comment by artificiel** [comment]
closed by #7465
